src/os/posix/src/os-impl-tasks.c

```diff
@@ -416,6 +424,24 @@ int32 OS_Posix_TaskAPI_Impl_Init(void)
    }
 #endif
 
+   pagesz = sysconf(_SC_PAGESIZE);
+   if (pagesz > 0)
+   {
+       /*
+        * convert to a mask.  Should already be a power of 2,
+        * but if not, make it so.
+        */
+       --pagesz;
+       pagesz |= (pagesz >> 1);
+       pagesz |= (pagesz >> 2);
+       pagesz |= (pagesz >> 4);
+       pagesz |= (pagesz >> 8);
+       pagesz |= (pagesz >> 16);
```

Comment:
But if it's not a power of two, won't it possibly fail the "may need to be a multiple" condition?  Why not leave as is so it won't possibly fail (and update the math to match)?

Reply:
Also, as the manpage, it said that it will only raise `EINVAL` on some system. I test it on my x86_64 i7 with `16385` (_SC_PAGESIZE = 4096), it didn't raise any error number to me.

Reply:
> But if it's not a power of two, won't it possibly fail the "may need to be a multiple" condition? Why not leave as is so it won't possibly fail (and update the math to match)?

Good point, shouldn't force to power of two.  Commit 43c48b2 changes to just making it a multiple of page size, whatever the number may be.

Reply:
>  as the manpage, it said that it will only raise `EINVAL` on some system

Right, and because we don't know whether or not the host system does or does not need this, we should use whole pages just in case.  On a virtual memory/MMU platform internally the memory allocation is almost certainly done in whole pages anyway.